Modules/Components/Examples/include/selxIdentityTransformRegistrationComponent.h

```diff
@@ -0,0 +1,105 @@
+
+#ifndef selxIdentityTransformRegistrationComponent_h
+#define selxIdentityTransformRegistrationComponent_h
+
+#include "selxSuperElastixComponent.h"
+#include "selxSinksAndSourcesInterfaces.h"
+#include "selxItkObjectInterfaces.h"
+
+#include "itkImageSource.h"
+
+namespace selx
+{
+template< int Dimensionality, class TPixel >
+class IdentityTransformRegistrationComponent :
+  public SuperElastixComponent<
+  Accepting<
+  itkImageFixedInterface< Dimensionality, TPixel >,
+  itkImageMovingInterface< Dimensionality, TPixel >
+  >,
+  Providing< RunRegistrationInterface,
+  DisplacementFieldItkImageSourceInterface< Dimensionality, TPixel >
+  >
+  >
+{
+public:
+
+  /** Standard ITK typedefs. */
+  typedef IdentityTransformRegistrationComponent<
+    Dimensionality, TPixel
+    >                                      Self;
+  typedef SuperElastixComponent<
+    Accepting<
+    itkImageFixedInterface< Dimensionality, TPixel >,
+    itkImageMovingInterface< Dimensionality, TPixel >
+    >,
+    Providing< RunRegistrationInterface,
+    DisplacementFieldItkImageSourceInterface< Dimensionality, TPixel >
+    >
+    > Superclass;
+  typedef std::shared_ptr< Self >       Pointer;
+  typedef std::shared_ptr< const Self > ConstPointer;
+
+  IdentityTransformRegistrationComponent( const std::string & name, LoggerImpl & logger );
+  virtual ~IdentityTransformRegistrationComponent();
+
+  typedef typename ComponentBase::CriterionType CriterionType;
+  typedef TPixel                                PixelType;
+
+  // fixed and moving image types are all the same, these aliases can be used to be explicit. 
+  using MovingImageType = typename itkImageMovingInterface< Dimensionality, TPixel >::ItkImageType;
+  using ResultImageType = typename itkImageInterface< Dimensionality, TPixel >::ItkImageType;
+  using DisplacementFieldImageType = typename DisplacementFieldItkImageSourceInterface< Dimensionality, TPixel >::ItkImageType;
+  
+  // Accepting Interfaces:
+  virtual int Accept( typename itkImageFixedInterface< Dimensionality, TPixel >::Pointer ) override;
+
+  virtual int Accept( typename itkImageMovingInterface< Dimensionality, TPixel >::Pointer ) override;
+
+  // Providing Interfaces:
+  virtual typename DisplacementFieldImageType::Pointer GetDisplacementFieldItkImage() override;
+
+  virtual void RunRegistration() override;
+
+  virtual bool MeetsCriterion( const CriterionType & criterion ) override;
+
+  static const char * GetDescription() { return "IdentityTransformRegistration Component"; }
+
+private:
+
+  typename DisplacementFieldImageType::Pointer m_DisplacementField;
+  typename itkImageFixedInterface< Dimensionality, TPixel >::Pointer m_ImageFixedInterface;
+  typename itkImageMovingInterface< Dimensionality, TPixel >::Pointer m_ImageMovingInterface;
+
+
+protected:
+
+  // return the class name and the template arguments to uniquely identify this component.
+  static inline const std::map< std::string, std::string > TemplateProperties()
```

Comment:
Should this type be aliased in the in the Superclass?

Reply:
That is indeed a style thing to discuss :) 
- A std::map< std::string, std::string > is very clear what it is, aliasing hides that
- In this specific case: TemplateProperties() is not in any base class (might be something we want). Now it is just a helper function (in principle can be any name and any type) for MeetsCriterion that components might want to have.